src/Vendor/Promotion/Rules/AndCatalogRule.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * @copyright Copyright (c) Wizacha
+ * @license Proprietary
+ */
+declare(strict_types=1);
+
+namespace Wizaplace\SDK\Vendor\Promotion\Rules;
+
+use Wizaplace\SDK\Vendor\Promotion\CatalogRuleType;
+
+/**
+ * Catalog promotion rule which is valid if all its items are valid.
+ */
+final class AndCatalogRule implements CatalogRule
```

Comment:
Pas fan du nom, mais `And` tout seul c'est refusé par PHP, et puis il y en aura une presque pareil mais légèrement différente pour les BasketRule